dag/dag.go

```diff
@@ -29,15 +29,14 @@ func (g *AcyclicGraph) DirectedGraph() Grapher {
 
 // Returns a Set that includes every Vertex yielded by walking down from the
 // provided starting Vertex v.
-func (g *AcyclicGraph) Ancestors(v Vertex) (*Set, error) {
-	s := new(Set)
-	start := AsVertexList(g.DownEdges(v))
+func (g *AcyclicGraph) Ancestors(v Vertex) (Set, error) {
+	s := make(Set)
```

Comment:
Go question -- make is for "type slice, map, or chan (only)" https://golang.org/pkg/builtin/#make. Why is this okay to use the dag package's `Set` type here?

Reply:
The underlying type of `Set` is `map[interface{}]interface{}`, so it is equivalent to calling `Set(make(map[interface{}]interface{}))`.  The same can be done with named slice and chan types (though the latter would be pretty unusual).

Reply:
Thanks!